d2l-input-checkbox.js

```diff
@@ -179,11 +191,23 @@ Polymer({
 	},
 	_handleChange: function(e) {
 		this.checked = e.target.checked;
+		this.indeterminate = false;
 		this.dispatchEvent(new CustomEvent(
 			'change',
 			{bubbles: true, composed: false}
 		));
 	},
+	_onClick: function() {
+		const browserType = window.navigator.userAgent;
+		if (this.indeterminate && (browserType.indexOf('Trident') > -1 || browserType.indexOf('Edge') > -1)) {
+			this.checked = !this.checked;
+			this.indeterminate = false;
+			this.dispatchEvent(new CustomEvent(
+				'change',
+				{bubbles: true, composed: false}
+			));
+		}
```

Comment:
Yikes, yeah I agree it sucks that this is necessary. It [seems](https://stackoverflow.com/questions/33523130/ie-does-not-fire-change-event-on-indeterminate-checkbox-when-you-click-on-it) | [like](https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/4710157/) | [others](https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/944049/) | [agree](https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/7344418/).

Reading around, apparently IE11 and Edge interpreted the specs differently than the other browsers: they're treating clicking the checkbox when it's indeterminate as simply removing the indeterminate state and therefore not worthy of a "change" event. The other browsers additionally toggle the checked state, which fires the change event. The specs aren't super clear what's correct.

My only concern with this hack is the same as any code that sniffs the user agent instead of using feature detection: in the near-ish future, Edge is going to start using Chromium. So at that point, the behaviour may flip for Edge. I don't know what their plans are with their user agent, but if they keep "Edge" in it this code will start failing. Having said that, after playing around with this I can't think of a non-user-agent-sniffing way to do this. 😭 

A couple suggested changes:
1. Throw a comment in here explaining why this code is necessary. Hopefully that'll help future-us untangle this when and if it falls apart.
2. If possible, this seems like a good place for some unit tests. I'd want to test checkboxes that are:
    - All combinations of initially `indeterminate`, initially checked, initially unchecked.
    - Initially not `indeterminate` that become `indeterminate` through both user interaction and by programatically setting the `indeterminate` property or removing the attribute (also both checked and unchecked)
    - In all cases, verify that the `checked` and `indeterminate` properties are initially correct and in the right state after "clicking". Also verify that the correct `change` events are fired.

Reply:
I completely agree that it's ugly. This took me a while to do, cuz I kept thinking there must be a better way. Apparently IE and Edge are "technically" correct, but people prefer how Chrome and others work.
I'll make those code changes. Good idea!